File: sdk/go/devices.go
package wlteopenapi

import (
	"context"
	"strconv"
)

type DevicesService struct {
	client *Client
}

func (s *DevicesService) List(ctx context.Context, options ListDevicesOptions) (DeviceList, error) {
	query := map[string]string{}
	if options.Page > 0 {
		query["page"] = strconv.Itoa(options.Page)
	}
	if options.PageSize > 0 {
		query["pageSize"] = strconv.Itoa(options.PageSize)
	}

	var result DeviceList
	err := s.client.request(ctx, "GET", "/wlte/v1/devices", query, nil, nil, &result)
	return result, err
}

func (s *DevicesService) Get(ctx context.Context, deviceID string) (Device, error) {
	var result Device
	err := s.client.request(ctx, "GET", "/wlte/v1/devices/"+urlEscape(deviceID), nil, nil, nil, &result)
	return result, err
}

func (s *DevicesService) GetConfig(ctx context.Context, deviceID string) (DeviceConfig, error) {
	var result DeviceConfig
	err := s.client.request(ctx, "GET", "/wlte/v1/devices/"+urlEscape(deviceID)+"/config", nil, nil, nil, &result)
	return result, err
}

func (s *DevicesService) Add(ctx context.Context, options AddDeviceOptions) (AddDeviceResult, error) {
	var result AddDeviceResult
	err := s.client.request(ctx, "POST", "/wlte/v1/devices", nil, nil, map[string]string{
		"deviceId": options.DeviceID,
		"password": options.Password,
		"name":     options.Name,
	}, &result)
	return result, err
}

func (s *DevicesService) Remove(ctx context.Context, deviceID string) (RemoveDeviceResult, error) {
	var result RemoveDeviceResult
	err := s.client.request(ctx, "DELETE", "/wlte/v1/devices/"+urlEscape(deviceID), nil, nil, nil, &result)
	return result, err
}

func (s *DevicesService) ModifyPassword(ctx context.Context, deviceID string, options ModifyDevicePasswordOptions) (ModifyDevicePasswordResult, error) {
	var result ModifyDevicePasswordResult
	err := s.client.request(ctx, "PUT", "/wlte/v1/devices/"+urlEscape(deviceID)+"/password", nil, nil, map[string]string{
		"oldPassword": options.OldPassword,
		"newPassword": options.NewPassword,
	}, &result)
	return result, err
}

File: sdk/typescript/src/devices.ts
import type { WlteClient } from './client.js'
import type {
  AddDeviceOptions,
  AddDeviceResult,
  ApiEnvelope,
  Device,
  DeviceConfig,
  DeviceList,
  ListDevicesOptions,
  ModifyDevicePasswordOptions,
  ModifyDevicePasswordResult,
  RemoveDeviceResult,
} from './types.js'

export class DevicesApi {
  constructor(private readonly client: WlteClient) {}

  async list(options: ListDevicesOptions = {}): Promise<DeviceList> {
    const response = await this.client.request<ApiEnvelope<DeviceList>>('/wlte/v1/devices', {
      query: {
        page: options.page,
        pageSize: options.pageSize,
      },
    })
    return response.data
  }

  async get(deviceId: string): Promise<Device> {
    const response = await this.client.request<ApiEnvelope<Device>>(`/wlte/v1/devices/${encodeURIComponent(deviceId)}`)
    return response.data
  }

  async getConfig(deviceId: string): Promise<DeviceConfig> {
    const response = await this.client.request<ApiEnvelope<DeviceConfig>>(
      `/wlte/v1/devices/${encodeURIComponent(deviceId)}/config`,
    )
    return response.data
  }

  async add(options: AddDeviceOptions): Promise<AddDeviceResult> {
    const response = await this.client.request<ApiEnvelope<AddDeviceResult>>('/wlte/v1/devices', {
      method: 'POST',
      body: options,
    })
    return response.data
  }

  async remove(deviceId: string): Promise<RemoveDeviceResult> {
    const response = await this.client.request<ApiEnvelope<RemoveDeviceResult>>(
      `/wlte/v1/devices/${encodeURIComponent(deviceId)}`,
      { method: 'DELETE' },
    )
    return response.data
  }

  async modifyPassword(deviceId: string, options: ModifyDevicePasswordOptions): Promise<ModifyDevicePasswordResult> {
    const response = await this.client.request<ApiEnvelope<ModifyDevicePasswordResult>>(
      `/wlte/v1/devices/${encodeURIComponent(deviceId)}/password`,
      { method: 'PUT', body: options },
    )
    return response.data
  }
}
